File: 122.买卖股票的最佳时机-ii.py
#
# @lc app=leetcode.cn id=122 lang=python
#
# [122] 买卖股票的最佳时机 II
#

# @lc code=start
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        arr1 = prices[0:-1]
        arr2 = prices[1:]
        return sum(map(lambda x, y: 0 if y - x <= 0 else y - x, arr1, arr2))
# @lc code=end

File: 122.买卖股票的最佳时机-ii.cpp
/*
 * @lc app=leetcode.cn id=122 lang=cpp
 *
 * [122] 买卖股票的最佳时机 II
 */

// @lc code=start
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        vector<int> res;
        for (int i = 0; i < prices.size() - 1; ++i) {
            auto tmp = prices[i + 1] - prices[i];
            if (tmp > 0) {
                res.push_back(tmp);
            }
        }
        return std::accumulate(res.begin(), res.end(), 0);
    }
};
// @lc code=end

